tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -844,39 +861,30 @@ def cwt_coefficients(x, c, param):
     :return: the different feature values
     :return type: pandas.Series
     """
-    df_cfg = pd.DataFrame(param)
-    res = pd.Series()
-
-    for widths in df_cfg["widths"].unique():
 
-        # the calculated_cwt with shape (len(widths), len(x)).
-        calculated_cwt = cwt(x, ricker, widths)
+    calculated_cwt = {}
+    res = []
+    indices = []
 
-        for w in df_cfg[df_cfg["widths"] == widths]["w"].unique():
+    for parameter_combination in param:
+        widths = parameter_combination["widths"]
+        w = parameter_combination["w"]
+        coeff = parameter_combination["coeff"]
 
-            # get the coefficients corresponding to this array of widths
-            coeff = df_cfg[(df_cfg["widths"] == widths) & (df_cfg["w"] == w)]["coeff"].unique()
+        if widths not in calculated_cwt:
```

Comment:
and faster, yes. This was the main intention :-)
All changes in this PR are just to make things faster (and cleaner, but this was just a side product).